oci_client.go

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"fmt"
+	"github.com/oracle/oci-go-sdk/common"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+)
+
+// OciClient stores the client and configuration details for making API requests to OCI Identity Service
+type OciClient struct {
+	common.BaseClient
+	config *common.ConfigurationProvider
+}
+
+// These constants store information related to signing the http request
+const (
+	// requestHeaderDate The key for passing a header to indicate Date
+	requestHeaderDate = "Date"
+
+	// requestHeaderUserAgent The key for passing a header to indicate User Agent
+	requestHeaderUserAgent = "User-Agent"
+
+	defaultScheme = "https"
+)
+
+// NewIdentityClientWithConfigurationProvider Creates a new default Identity client with the given configuration provider.
+// the configuration provider will be used for the default signer as well as reading the region
+func NewOciClientWithConfigurationProvider(configProvider common.ConfigurationProvider) (client OciClient, err error) {
+	baseClient, err := common.NewClientWithConfig(configProvider)
+	if err != nil {
+		return
+	}
+
+	client = OciClient{BaseClient: baseClient}
+	client.BasePath = ""
+	err = client.setConfigurationProvider(configProvider)
```

Comment:
This will also not be the `err` in your function signature, because of line 34.

Reply:
Done